addons/binding/org.openhab.binding.sensebox/src/main/java/org/openhab/binding/sensebox/handler/SenseBoxHandler.java

```diff
@@ -0,0 +1,374 @@
+                        thingConfiguration.getSenseBoxId(), thingConfiguration.getRefreshInterval());
+
+                try {
+                    boolean success = fetchData();
+                    if (success) {
+                        publishDataForChannel(new ChannelUID(getThing().getUID(), CHANNEL_NAME));
+                        publishDataForChannel(new ChannelUID(getThing().getUID(), CHANNEL_LOCATION));
+                        publishDataForChannel(new ChannelUID(getThing().getUID(), CHANNEL_EXPOSURE));
+                        publishDataForChannel(new ChannelUID(getThing().getUID(), CHANNEL_API_URL));
+                        publishDataForChannel(new ChannelUID(getThing().getUID(), CHANNEL_MAP_URL));
+                        publishDataForChannel(new ChannelUID(getThing().getUID(), CHANNEL_IMAGE_URL));
+
+                        publishDataForChannel(new ChannelUID(getThing().getUID(), CHANNEL_UV_INTENSITY));
+                        publishDataForChannel(new ChannelUID(getThing().getUID(), CHANNEL_LUMINANCE));
+                        publishDataForChannel(new ChannelUID(getThing().getUID(), CHANNEL_PRESSURE));
+                        publishDataForChannel(new ChannelUID(getThing().getUID(), CHANNEL_HUMIDITY));
+                        publishDataForChannel(new ChannelUID(getThing().getUID(), CHANNEL_TEMPERATURE));
+
+                        publishDataForChannel(new ChannelUID(getThing().getUID(), CHANNEL_UV_INTENSITY_LR));
+                        publishDataForChannel(new ChannelUID(getThing().getUID(), CHANNEL_LUMINANCE_LR));
+                        publishDataForChannel(new ChannelUID(getThing().getUID(), CHANNEL_PRESSURE_LR));
+                        publishDataForChannel(new ChannelUID(getThing().getUID(), CHANNEL_HUMIDITY_LR));
+                        publishDataForChannel(new ChannelUID(getThing().getUID(), CHANNEL_TEMPERATURE_LR));
+                    }
+                } catch (Exception e) {
+                    logger.debug("Exception occurred during execution: {}", e.getMessage(), e);
+                }
+            }
+        };
+
+        refreshJob = scheduler.scheduleAtFixedRate(runnable, 0, thingConfiguration.getRefreshInterval(),
+                TimeUnit.SECONDS);
+    }
+
+    private boolean fetchData() {
+        try {
+            data = CACHE.get(thingConfiguration.getSenseBoxId());
+        } catch (IOException e) {
+            logger.debug("Exception occurred during fetching data: {}", e.getMessage(), e);
+            return false;
+        }
+
+        return true;
+    }
+
+    private final ExpiringCache<String, SenseBoxData> CACHE = new ExpiringCache<>(CACHE_EXPIRY,
+            new ExpiringCache.LoadAction<String, SenseBoxData>() {
+                @Override
+                public SenseBoxData load(String senseBoxId) throws IOException {
+                    SenseBoxData result = reallyFetchDataFromServer(senseBoxId);
```

Comment:
True :)

But in such cases, I actually prefer to have the local variable, as it helps in the debugger, and for productive code, the compiler optimizes it anyway 😄